src/js/disability-benefits/526EZ/config/form.js

```diff
@@ -409,9 +424,22 @@ const formConfig = {
                     viewField: treatmentView
                   },
                   items: {
-                    treatmentCenterName: {
-                      'ui:title': 'Name of VA medical facility'
-                    },
+                    treatmentCenterName: autoSuggestUiSchema(
+                      'Name of VA medical facility',
+                      // Wait for 1 second before querying
+                      // It's debatable tha we want this to be _.throttle instead of _.debounce
+                      // The 1 second is super arbitrary
+                      _.debounce(1000, queryForFacilities),
```

Comment:
I kinda feel like this should be in the field component, instead of leaving it to someone creating a form. It's generally a good practice to not fire a request on every key stroke.

Reply:
Yeah, I left it out here because this callback isn't _necessarily_ an AJAX request. I can't think of any other legitimate use cases, but I opted for the most flexibility.

That said, I'm not opinionated. I can make the debounce wait time a configurable option. If it's then set to 0, it's _basically_ like it's not debounced at all, which solves that problem if anybody really needs it solved.

Good point. I'll move it. :+1: